### DISCUSSION TITLE
How to receive system events

### DISCUSSION THREAD

**Discussion by kockie69** [mpaperno/WASimCommander#15] [Q&A]
Hi,

It is not clear to me , and I couldn't find an example, how to receive System Events. I am for example waiting for a SIMCONNECT_RECV_ID_EVENT_FRAME event. Is there a way to keep receiving these?

I might be missing the trick, but it feels to me these events are filtered somewhere down the line and won't make it to the client anymore.

Thanks for your help.

**Comment by mpaperno** [comment]
Hi,

WASim doesn't pass through the SimConnect messages at all. SimConnect is only used internally as the network transport, and the WASimClient doesn't expose the SimConnect instance it uses "behind the scenes."
(Actually I'd like to get rid of SC entirely, but there's no other way to network with the sim.)

It is not really designed to replace or replicate SimConnect, though in some cases it can/does (in terms of requesting vars and firing sim events).  I'm open to expanding the functionality, if it makes sense.

May I ask what you are trying to use the sim frame event for?  Or WASimCommander in general.

Cheers,
-Max

**Reply by kockie69** [reply]
I am working on a POC to use MSFS as a VR cockpit and scenery generator for PSX (Aerowinx). So I was looking if your solution could make my live easier.
